tmk_core/protocol/vusb/vusb.c

```diff
@@ -48,40 +48,38 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #    include "os_detection.h"
 #endif
 
-#define NEXT_INTERFACE __COUNTER__
-
 /*
  * Interface indexes
  */
 enum usb_interfaces {
 #ifndef KEYBOARD_SHARED_EP
-    KEYBOARD_INTERFACE = NEXT_INTERFACE,
+    KEYBOARD_INTERFACE,
 #else
-    SHARED_INTERFACE = NEXT_INTERFACE,
+    SHARED_INTERFACE,
 #    define KEYBOARD_INTERFACE SHARED_INTERFACE
 #endif
 
 // It is important that the Raw HID interface is at a constant
 // interface number, to support Linux/OSX platforms and chrome.hid
 // If Raw HID is enabled, let it be always 1.
 #ifdef RAW_ENABLE
-    RAW_INTERFACE = NEXT_INTERFACE,
+    RAW_INTERFACE,
 #endif
 
 #if defined(SHARED_EP_ENABLE) && !defined(KEYBOARD_SHARED_EP)
-    SHARED_INTERFACE = NEXT_INTERFACE,
+    SHARED_INTERFACE,
 #endif
 
 #ifdef CONSOLE_ENABLE
-    CONSOLE_INTERFACE = NEXT_INTERFACE,
+    CONSOLE_INTERFACE,
 #endif
 
-    TOTAL_INTERFACES = NEXT_INTERFACE
+    TOTAL_INTERFACES
 };
 
 #define MAX_INTERFACES 3
 
-#if (NEXT_INTERFACE - 1) > MAX_INTERFACES
+#if TOTAL_INTERFACES > MAX_INTERFACES
 #    error There are not enough available interfaces to support all functions. Please disable one or more of the following: Mouse Keys, Extra Keys, Raw HID, Console
 #endif
```

Comment:
This check no longer works, because the value of `TOTAL_INTERFACES` is not known to the preprocessor (but the preprocessor silently replaces all undefined identifiers with 0 when evaluating expressions, so the condition evaluates as false).  Checking the value of a enum member would require a static assert instead.
```suggestion
_Static_assert(TOTAL_INTERFACES <= MAX_INTERFACES, "There are not enough available interfaces to support all functions. Please disable one or more of the following: Mouse Keys, Extra Keys, Raw HID, Console.");
```